### PR TITLE
group_index::explore_groups + supporting changes

### PR DESCRIPTION
**Pull request by megrogan**

### CODE REVIEW

File: backend/canisters/group/impl/src/lifecycle/post_upgrade.rs [hunk 1]
```diff
@@ -1,13 +1,17 @@
+use std::time::Duration;
+
```
Comment (hpeebles) [inline backend/canisters/group/impl/src/lifecycle/post_upgrade.rs:2]: Unnecessary whitespace

File: backend/canisters/group/impl/src/lifecycle/post_upgrade.rs [hunk 2]
```diff
@@ -1,13 +1,16 @@
+use std::time::Duration;
```
Comment (hpeebles) [inline backend/canisters/group/impl/src/lifecycle/post_upgrade.rs:1]: This will fail fmt

File: backend/canisters/group_index/impl/src/jobs/mod.rs [hunk 3]
```diff
@@ -1,7 +1,7 @@
 use crate::RuntimeState;
 
-mod calculate_community_hotness;
 mod calculate_hot_groups;
```
Comment (hpeebles) [inline backend/canisters/group_index/impl/src/jobs/mod.rs:3]: Can the `calculate_hot_groups` job be deleted now?

Comment (megrogan) [reply backend/canisters/group_index/impl/src/jobs/mod.rs:3]: Once the FE uses explore_groups we can of course delete "search" and "recommended_groups" and all supporting code including the calculate_hot_groups job.

**Review by hpeebles — APPROVED** [review]
